Content.Client/Remotes/UI/DoorRemoteStatusControl.cs

```diff
@@ -0,0 +1,46 @@
+using Content.Client.Message;
+using Content.Client.Stylesheets;
+using Content.Shared.Remotes.Components;
+using Robust.Client.UserInterface;
+using Robust.Client.UserInterface.Controls;
+using Robust.Shared.Timing;
+
+namespace Content.Client.Remote.UI;
+
+public sealed class DoorRemoteStatusControl : Control
+{
+    private readonly Entity<DoorRemoteComponent> _entity;
+    private readonly RichTextLabel _label;
+
+    // set to toggle bolts initially just so that it updates on first pickup of remote
+    private OperatingMode PrevOperatingMode = OperatingMode.placeholderForUiUpdates;
+
+    public DoorRemoteStatusControl(Entity<DoorRemoteComponent> entity)
+    {
+        _entity = entity;
+        _label = new RichTextLabel { StyleClasses = { StyleNano.StyleClassItemStatus } };
+        AddChild(_label);
+    }
+
+    protected override void FrameUpdate(FrameEventArgs args)
+    {
+        base.FrameUpdate(args);
+
+        // only updates the UI if any of the details are different than they previously were
+        if (PrevOperatingMode == _entity.Comp.Mode)
+            return;
+
+        PrevOperatingMode = _entity.Comp.Mode;
+
+        // Update current volume and injector state
+        var modeStringLocalized = Loc.GetString(_entity.Comp.Mode switch
+        {
+            OperatingMode.OpenClose => "door-remote-open-close-text",
+            OperatingMode.ToggleBolts => "door-remote-toggle-bolt-text",
+            OperatingMode.ToggleEmergencyAccess => "door-remote-emergency-access-text",
+            _ => "door-remote-invalid-text"
+        });
+
+        _label.SetMarkup(Loc.GetString("door-remote-mode-label", ("modeString", modeStringLocalized)));
+    }
```

Comment:
You don't need to be checking the mode every single frame, just update it when needed.

Reply:
darn
I already did it that way for the hypospray and injectors, so I guess it's time to figure out a fix for all three...

Reply:
and it turns out every single other status control also does a check every frameupdate...... so I guess I need to figure this out for every single status control

Reply:
> You don't need to be checking the mode every single frame, just update it when needed.

It's not universally possible to do this. Lightweight polling for a UI control that is at most visible twice at once is fine.

Reply:
I understand that "AfterAutoHandleStateEvent" is a thing, but I still don't know how to get the DoorRemoteStatusControl class to be aware of the updates since I can only subscribe to it with the DoorRemoteSystem... to my knowledge?

> It's not universally possible to do this. Lightweight polling for a UI control that is at most visible twice at once is fine.
oh....

Reply:
> does that mean I don't have to change this...?

Not in my opinion, no

Reply:
> > You don't need to be checking the mode every single frame, just update it when needed.
> 
> It's not universally possible to do this. Lightweight polling for a UI control that is at most visible twice at once is fine.

What's wrong with just using the afterstatehandler or adding a method to set mode that also updates the control?

Reply:
> > > You don't need to be checking the mode every single frame, just update it when needed.
> > 
> > 
> > It's not universally possible to do this. Lightweight polling for a UI control that is at most visible twice at once is fine.
> 
> What's wrong with just using the afterstatehandler or adding a method to set mode that also updates the control?

Currently the only thing stopping me is my lack of knowledge on how to add to the Control label without using Subs.ItemStatus and another class that inherits Control...

Reply:
> What's wrong with just using the afterstatehandler or adding a method to set mode that also updates the control?

1. it's annoying
2. the cost of polling doesn't matter
3. it's not universally applicable to all components anyways, more complex stuff can't do it